newrelic/core/agent_control_health.py

```diff
@@ -216,7 +215,15 @@ def agent_control_health_instance():
 
 
 def agent_control_healthcheck_loop():
-    reporting_frequency = os.environ.get("NEW_RELIC_AGENT_CONTROL_HEALTH_FREQUENCY", 5)
+    try:
+        reporting_frequency = int(os.environ.get("NEW_RELIC_AGENT_CONTROL_HEALTH_FREQUENCY", 5))
+        # If we have an invalid integer value for frequency, default back to 5
+        if reporting_frequency <= 0:
+            reporting_frequency = 5
+    except Exception:
+        # If we run into an exception when int typecasting, default back to 5
+        reporting_frequency = 5
```

Comment:
This one could probably be `_environ_as_int()` with a default of 5, and then run this check afterwards.

```
        if reporting_frequency <= 0:
            reporting_frequency = 5
```